### ISSUE TITLE
Mouseup event query - additional event fires with evt.type = 'mouseup' but  evt.evt.type='click' 

### ISSUE THREAD

**Issue by VanquishedWombat** [konvajs/konva#1390]
I may be misunderstanding the seq of events but....

tldr: demo:  https://codepen.io/JEE42/pen/QWmVBOe

### Description 
if I make a rect on a stage and add the following 

``` 
rect.on('mousedown mouseup click', function(evt){
  console.log('rect listener triggers event named [' + evt.evt.type +'] with evt.type =[' + evt.type + ']')
})
```

and I mousedown - hold for 2 secs - then mouseup,

### What did I expect ? 

I expected 2 lines as follows:
- rect listener triggers event named [mousedown] with evt.type =[mousedown]
- rect listener triggers event named [mouseup] with evt.type =[mouseup]

### What did I get ?

- rect listener triggers event named [mousedown] with evt.type =[mousedown]
- rect listener triggers event named [mouseup] with evt.type =[mouseup]
- rect listener triggers event named [**mouseup**] with evt.type =[**click**]

### Why is this wrong ?

The first 2 lines are expected. Those are the events I triggered. But the third line confuses me for 2 reasons:

1 - why did it get triggered?
2 - the evt.evt object has it's own property 'type', which says 'mouseup' but then there is the evt.type which says 'click'.  This inconsistency is confusing.

### What should we do ?

Either stop the third event from happening, or make the evt.type and evt.evt.type line up.  Update the docs to clearly explain that if users are wanting to only have the mouseup.mouseup  event handled then they need to filter out the click.mouseup event. 

Like I said, I may be misunderstanding something - I _get_ that the click and mousedown-mouseup combo are closely related and so this effect may be something to do with that.

**Comment by lavrton** [comment]
> 1 - why did it get triggered?

Click is triggered, no matter how long you wait. The DOM `click` works the same way.

> 2 - the evt.evt object has it's own property 'type', which says 'mouseup' but then there is the evt.type which says 'click'. This inconsistency is confusing.

It is specific to implementation of Konva event system. In short, Konva is listening only to `mousedown`, `mousemove` and `mouseup` events. All other events like `mouseleave`, `click` and `dblclick` are produced from these events. So `click` is `mousedown` + `mouseup` (without dragging). `evt.evt` is a native event object, just for reference. Usually, developers don't need to use it. This is why it still has `mouseup` type, while Konva fired `click` event.

It will be better to stay closer to native events. But it is not always possible (e.g. `mouseleave` event on shape will be still native `mousemove`). But I am not sure it is very important.

**Comment by VanquishedWombat** [comment]
Ok - all makes sense. So we should use the evt.type and not the evt.evt.type in our Konva dev work. Understood

Please close the issue.
